rnaseq-cgl-pipeline/rnaseq-cgl-pipeline.cwl

```diff
@@ -0,0 +1,142 @@
+    Kallisto: abundance.tsv, abundance.h5, and a JSON of run information
+    ```
+
+    *Feedback*
+
+    If there are any questions please contact the workflow author John Vivian (jtvivian@gmail.com). If you find any errors or corrections please feel free to make a pull request. Feedback of any kind is appreciated.
+
+
+dct:creator:
+  '@id': http://orcid.org/0000-0002-7681-6415
+  foaf:name: Brian O'Connor
+  foaf:mbox: briandoconnor@gmail.com
+
+requirements:
+  - class: DockerRequirement
+    dockerPull: "quay.io/briandoconnor/rnaseq-cgl-pipeline:2.0.10-3"
+
+hints:
+  - class: ResourceRequirement
+    coresMin: 1
+    ramMin: 64000
+    outdirMin: 500000000
+    description: "The process requires at least 16G of RAM and we recommend 500GB or storage."
+
+inputs:
+  samples:
+    doc: "Absolute path(s) to sample tarballs or paired FASTQ files. FASTQ pairs are comma delimited and each pair is space delimited. Ex: sample1,sample2 sample3,sample4"
+    type:
+      type: array
+      items: File
+    inputBinding:
+      prefix: --samples
+
+  star:
+    type: File
+    doc: "Absolute path to STAR index tarball."
+    inputBinding:
+      prefix: --star
+
+  rsem:
+    type: File
+    doc: "Absolute path to rsem reference tarball."
+    inputBinding:
+      prefix: --rsem
+
+  kallisto:
+    type: File
+    doc: "Absolute path to kallisto index (.idx) file."
+    inputBinding:
+      prefix: --kallisto
+
+  disable-cutadapt:
+    type: boolean?
+    default: false
+    doc: "Cutadapt fails if samples are improperly paired. Use this flag to disable cutadapt."
+    inputBinding:
+      prefix: --disable-cutadapt
+
+  save-bam:
+    type: boolean?
+    default: false
+    doc: "If this flag is used, genome-aligned bam is written to output."
+    inputBinding:
+      prefix: --save-bam
+
+  save-wiggle:
+    type: boolean?
+    default: false
+    doc: "If this flag is used, wiggle files (.bg) are written to output."
+    inputBinding:
+      prefix: --save-wiggle
+
+  bamqc:
+    type: boolean?
+    default: false
+    doc: "If this flag is used, the BAMQC step will be run."
+    inputBinding:
+      prefix: --bamqc
+
+  work-mount:
+    type: string
+    doc: "Path of the working directory to be mounted into the container"
+    inputBinding:
+      prefix: --work_mount
+
+  no-clean:
+    type: boolean?
+    default: true
+    doc: "If this flag is used, temporary work directory is not cleaned."
+    inputBinding:
+      prefix: --no-clean
+
+  resume:
+    type: string?
+    doc: "Path of the working directory that contains a job store to be resumed."
+    inputBinding:
+      prefix: --resume
+
+  cores:
+    type: int?
+    doc: "Will set a cap on number of cores to use, default is all available cores."
+    inputBinding:
+      prefix: --cores
+
+outputs:
+  output_files:
+    type:
+      type: array
+      items: File
+    outputBinding:
+      glob: '*.tar.gz'
+    doc: "Result files RNA-seq CGL pipeline"
+
```

Comment:
Hi CJ,
Please add these two extra output file arrays to the outputs: section of the CWl so the whole outputs: section will look like this:
outputs:
  output_files:
    type:
      type: array
      items: File
    outputBinding:
      glob: '*.tar.gz'
    doc: "Result files RNA-seq CGL pipeline"

  wiggle_files:
    type:
      type: array
      items: File
    outputBinding:
      glob: '*.wiggle.bg'
    doc: "Wiggle result files RNA-seq CGL pipeline"

  bam_files:
    type:
      type: array
      items: File
    outputBinding:
      glob: '*.bam'
    doc: "BAM result files RNA-seq CGL pipeline"